projects/plugins/crm/api/create_transaction.php

```diff
@@ -107,57 +125,33 @@
 				);
 
 }
-
-		// RECOMMENDED
-		$status = '';
-if ( isset( $new_trans['status'] ) ) {
-	$status = sanitize_text_field( $new_trans['status'] );
-}
-		$total = '';
-if ( isset( $new_trans['total'] ) ) {
-	$total = sanitize_text_field( $new_trans['total'] );
-}
-		$item_title = '';
-if ( isset( $new_trans['item_title'] ) ) {
-	$item_title = sanitize_text_field( $new_trans['item_title'] );
-}
-		$net = '';
-if ( isset( $new_trans['net'] ) ) {
-	$net = sanitize_text_field( $new_trans['net'] );
-}
-		$tax = '';
-if ( isset( $new_trans['tax'] ) ) {
-	$tax = sanitize_text_field( $new_trans['tax'] );
-}
-		$fee = '';
-if ( isset( $new_trans['fee'] ) ) {
-	$fee = sanitize_text_field( $new_trans['fee'] );
-}
-		$disc = '';
-if ( isset( $new_trans['disc'] ) ) {
-	$disc = sanitize_text_field( $new_trans['discount'] );
-}
-		$rate = '';
-if ( isset( $new_trans['rate'] ) ) {
-	$rate = sanitize_text_field( $new_trans['tax'] );
-}
-		$date = isset( $new_trans['date'] ) ? date( 'Y-m-d H:i:s', (int) $new_trans['date'] ) : '';
-
-		$tFields = array(
+		$transaction_fields = array(
 			'orderid'  => $orderid,
 			'customer' => $customer,
-			'status'   => $status,
-			'total'    => $total,
-			'date'     => $date,
-
-			'item'     => $item_title,
-			'net'      => $net,
-			'tax'      => $tax,
-			'fee'      => $fee,
-			'discount' => $disc,
-			'tax_rate' => $rate,
+			'status'   => '',
+			'total'    => '',
+			'date'     => '',
+			'item'     => '',
+			'net'      => '',
+			'tax'      => '',
+			'fee'      => '',
+			'discount' => '',
+			'tax_rate' => '',
 		);
 
+		foreach ( $new_trans as $field => $value ) {
+
+			if ( $field === 'orderid' || $field === 'email' || $field === 'fname' ) {
+				continue;
+			}
+			if ( $field === 'date' ) {
+				$transaction_fields[ $field ] = gmdate( 'Y-m-d H:i:s', (int) $value );
+				continue;
+			}
+
+			$transaction_fields[ $field ] = sanitize_text_field( $value );
```

Comment:
Given the purpose is to handle custom fields, this will not work unless it's a text-type field. Some expect numbers, arrays (or CSV, I can't remember), textarea...in particular, textarea is finicky here.

I believe the "correct" way to handle this would be to use `zeroBS_buildObjArr()` (which has its own problems, but should work in this scenario to sanitise and such).

Reply:
> Given the purpose is to handle custom fields, this will not work unless it's a text-type field. Some expect numbers, arrays (or CSV, I can't remember), textarea...in particular, textarea is finicky here.

Another potential issue for the future... For now, it manages well the most custom field types, all the basic ones: text, numbers, and also textareas. I don't see any issue here.